### ISSUE TITLE
Fix Dockerfile

### ISSUE THREAD

**Issue by micorix** [po8klasie/po8klasie-rails-api#23]
After upgrading image version to `ruby:3.1.0-alpine` Dockerfile got broken.

1. Current Dockerfile on master branch include `apt-get` instructions, which indicate debian-based image. We use alpine version. Pls change to `RUN apk update && apk add ...`. New RUN instruction also lack some deps, was that intended?
2. Building an image throws following error: `Fetching https://github.com/rswag/rswag.git. You need to install git to be able to use gems from git repositories.` We might want to install git or use debian-based image which (afaik) comes with git preinstalled

These errors show that having the code tested in GitHub Actions by building the image and running tests in containers would be helpful.

@Anakin100100 Please fix dockerfile and make sure it runs fine. Then I'll prepare a PR for running docker in gh actions.

**Comment by Anakin100100** [comment]
The dockerfile on master is based on ubuntu image. This is the most general use and most consistent in development experience. Short story is that I first switched to debian-buster-slim because there were severe issues with alpine DNS and later I switched from debian-buster-slim to the ubuntu based one because there were severe issues with the postgres binaries. This simply works and we won't be going back to alpine. 

The dockerfile on master is perfectly fine. There was a small issue with the rswag gem version because for a short time we needed to check it out from github but the newest release fixed the issue. After we merge data_filtering any issues will be resolved. If you want to merge the dockerfile branch faster change the line with gem 'rswag' .... to only gem 'rswag'

We cannot run the tests inside the docker image because the dockerfile defines a production env where the testing and development dependencies aren't installed. The issues in dockerfile ussually arise from it being in production mode and rarely from unit tests failing.
